trino/client.py

```diff
@@ -556,6 +581,18 @@ def process(self, http_response) -> TrinoStatus:
         if constants.HEADER_SET_ROLE in http_response.headers:
             self._client_session.role = http_response.headers[constants.HEADER_SET_ROLE]
 
+        if constants.HEADER_ADDED_PREPARE in http_response.headers:
+            for name, statement in get_prepared_statement_values(
+                http_response.headers, constants.HEADER_ADDED_PREPARE
+            ):
+                self._client_session.prepared_statements[name] = statement
+
+        if constants.HEADER_DEALLOCATED_PREPARE in http_response.headers:
+            for name in get_header_values(
+                http_response.headers, constants.HEADER_DEALLOCATED_PREPARE
+            ):
+                self._client_session.prepared_statements.pop(name)
+
```

Comment:
Can you please add tests for setting proper headers in ClientSession?

Reply:
ClientSession doesn't store header values - it stores information read from headers.

Do you mean a test that a request which includes ADDED header actually affects the `prepared_statements` dict and so on? Isn't that covered by the prepared statements tests themselves?

Reply:
Yes, you are right. But I don't think those tests are there. I can only see `test_trino_query_response_headers` in `test_client.py`.

Reply:
They live in test_dbapi_integration.py and test_sqlalchemy_integration.py. See the various `_params` test methods.

I can add some tests verifying that TrinoRequest#http_headers returns correct headers when prepared statements exist in ClientSession but testing the "implementation" is generally not any more useful than testing the "feature".

Let me know what you'd prefer.

Reply:
It makes more sense to test the "feature" in this case.  

Disscused offline to add a test to assert that after each query the `prepared_statements` dict is empty (to make sure we don't leak things)